lib/formatters/json.js

```diff
@@ -0,0 +1,20 @@
+/**
+ * @fileoverview JSON reporter
+ * @author Burak Yigit Kaya aka BYK
+ * @copyright 2015 Burak Yigit Kaya. All rights reserved.
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Public Interface
+//------------------------------------------------------------------------------
+
+module.exports = function(results) {
+    var output = {};
+
+    results.forEach(function(result) {
+        output[result.filePath] = (output[result.filePath] || []).concat(result.messages);
+    });
+
+    return JSON.stringify(output);
```

Comment:
Since the original ticket mentioned piping this to other programs, I didn't consider human readability, at least for an initial implementation.

Also, I think there are already much better formatters in the project if one is concerned about human readability :D